app/server/utils/ussd/kenya_ussd_processor.py

```diff
@@ -42,41 +44,60 @@ def next_state(session: UssdSession, user_input: str, user: User) -> UssdMenu:
         return new_state
 
     @staticmethod
-    def replace_vars(menu: UssdMenu, ussd_session: UssdSession, display_text: str, user: User) -> str:
-        replacements = [['%support_phone%', '+254757628885']]
+    def custom_display_text(menu: UssdMenu, ussd_session: UssdSession, user: User) -> Optional[str]:
 
         if menu.name == 'about_my_business':
-            replacements.append(['%user_bio%', user.bio])
-        elif menu.name == 'send_token_confirmation':
-            phone = proccess_phone_number(ussd_session.session_data['recipient_phone'], 'KE')
-            recipient = User.query.filter_by(phone=phone).first()
-            replacements.append(['%recipient_phone%', recipient.user_details()])
-            # TODO(ussd): this is not a thing yet!!
-            token = ussd_session.user.community_token
-            replacements.append(['%token_name%', token.name])
-            replacements.append(['%transaction_amount%', ussd_session.session_data['transaction_amount']])
-            replacements.append(['%transaction_reason%', ussd_session.session_data['transaction_reason']])
-        elif menu.name == 'exchange_token_confirmation':
-            phone = proccess_phone_number(ussd_session.session_data['agent_phone'], 'KE')
-            agent = User.query.filter_by(phone=phone).first()
-            replacements.append(['%agent_phone%', agent.user_details])
-            # TODO(ussd): this is not a thing yet!!
-            token = ussd_session.user.community_token
-            replacements.append(['%token_name%', token.name])
-            replacements.append(['%exchange_amount%', ussd_session.session_data['exchange_amount']])
-        elif 'pin_authorization' in menu.name or 'current_pin' in menu.name:
+            bio = user.custom_attributes.filter_by(name='bio').first()
+            if bio is None:
+                # TODO: replace this with a no bio message?
+                return i18n_for(user, menu.display_key, user_bio=bio)
+            else:
+                return i18n_for(user, menu.display_key, user_bio=bio)
+
+        if menu.name == 'send_token_confirmation':
+            recipient = get_user_by_phone(ussd_session.get_data('recipient_phone'), 'KE', True)
+            recipient_phone = recipient.user_details()
+            token = default_token(user)
+            transaction_amount = ussd_session.get_data('transaction_amount')
```

Comment:
I'm getting:

```
  File "/Users/Nick/PycharmProjects/SempoBlockchain/app/server/utils/ussd/kenya_ussd_processor.py", line 61, in custom_display_text
    transaction_amount = ussd_session.get_data('transaction_amount')
  File "/Users/Nick/PycharmProjects/SempoBlockchain/app/server/models/ussd.py", line 54, in get_data
    return self.session_data[key]
KeyError: 'transaction_amount'

```

Reply:
is this when you run it for real? that is still completely untested, will get to it

Reply:
ohh yeah it was me trying to dry run with ussd lol